website/blog/2018-06-25-on-consuming-and-publishing-es6+-packages.md

```diff
@@ -0,0 +1,276 @@
+
+For Babel, we may need to add some feature requests to `@babel/cli` to make this easier, and maybe make the `babel` package do this by default? Or we should integrate better with tools like @developit's [microbundle](https://github.com/developit/microbundle).
+
+And how do we deal with polyfills (this will be an upcoming post)? What would it look like for a library author (or the user) to not to have to think about polyfills?
+
+With all that said, how does Babel help with all this?
+
+## How Babel v7 Helps
+
+As we've discussed, compiling dependencies in Babel v6 can be pretty painful. Babel v7 will address some of these pain points. 
+
+One issue is around configuration lookup. Babel currently runs per file, so when compiling a file, it tries to find the closest config ([`.babelrc`](https://babeljs.io/docs/en/next/babelrc)) to know what to compile against. It keeps looking up the directory tree if it doesn't find it in the current folder.
+
+```
+project
+└── .babelrc // closest config for a.js
+└── a.js
+└── node_modules
+    └── package
+        └── .babelrc // closest config for b.js
+        └── b.js
+```
+
+We made a [few changes](https://github.com/babel/babel/pull/7358):
+
+- One is to stop lookup at the package boundary (stop when we find a `package.json`). This makes sure Babel won't try to load a config file outside the app, the most surprising being with it finds one in your home directory.
+- If we use a monorepo, we may want to have a `.babelrc` per-package that extends some other central config.
+- Babel itself is a monorepo, so instead we are using the new `babel.config.js` which allows us to resolve all files to that config (no more lookup).
+- We added an [`"overrides"`](https://github.com/babel/babel/pull/7091) option which allows us to basically create a new config for any set of paths.
+
+This allows us to have a single config for our whole app: maybe we want to compile our server JavaScript code differently than the client code as well as compile something in `node_modules`.
+
+```js
+// babel.config.js
+module.exports = {
+  presets: [
+    ['env', { 
+      targets: { node: 'current' },
+    }],
+  ],
+  overrides: [{
+    test: "./client",
+    presets: [
+      ['env', { 
+        targets: { browsers: ["last 2 versions"] },
+      }],
+    ],
+  }, {
+    test: "./node_modules/",
+    presets: [
+      ['env', { 
+        targets: { browsers: ["last 2 versions"] },
+      }],
+    ],
+  }],
+}
+```
+
+## Recommendations to Discuss
+
+We should shift our fixed view of publishing JavaScript to one that keeps up with the standard which pushes forward.
+
+My recommendation is that package authors should continue to publish ES5/CJS under `main` for backwards compat with current tooling and workflow but also publish a version compiled down to latest syntax (no experimental proposals) under a new key we can standardize on like `main-es`. I don't believe `module` should be that key.
```

Comment:
Why not?

Reply:
Maybe link back to the “Conflating ESM and ES2015+” section or reiterate the take-aways here. (I’d prefer the latter)